examples/stream/stream.mjs

```diff
@@ -0,0 +1,103 @@
+import {
+  Writable,
+  Readable,
+  Duplex
+} from 'stream';
+
+const kPort = Symbol('kPort');
+
+export class MessagePortWritable extends Writable {
+  constructor (port, options) {
+    super(options);
+    this[kPort] = port;
+  }
+
+  _write (buf, _, cb) {
+    this[kPort].postMessage(buf, [buf.buffer]);
+    cb();
+  }
+
+  _writev (data, cb) {
+    for (let n = 0; n < data.length; n++) {
+      this[kPort].postMessage(data[n].chunk, [data[n].chunk.buffer]);
```

Comment:
I'd recommend to add batch support and pass the array through as-is.

Reply:
Heh, yeah, I added that to the duplex, forgot to update the Writable.